src/backend/ticket_service/ticket_service.py

```diff
@@ -376,4 +377,262 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
+            comment.posted_by,
+            comment.comment
+        )
+
+        logger.debug(f"Database response: {db_response}")
+        return CommentResponse(comment_id=db_response["comment_id"], comment=db_response["comment"])
+
+    except ValueError as e:
+        logger.error(f"Validation error: {e}")
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail=str(e))
+
+    except OperationalError as e:
+        logger.error(f"Operational error: {e}")
+        raise HTTPException(
+            status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+            detail="The database is currently unreachable. Please try again later."
+        )
+    except DatabaseError as e:
+        logger.error(f"Database error: {e}")
+        raise HTTPException(
+            status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+            detail="The database is currently unreachable. Please try again later."
+        )
+    except SQLAlchemyError as e:
+        logger.error(f"Database error: {e}")
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Database error occurred")
+
+    except Exception as e:
+        logger.error(f"Unexpected error in creating comment: {e}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail="An unexpected error occurred"
+        )
+
+
+@app.get("/tickets/{tickets_id}/comments/{comment_id}", response_model=Comment, tags=["Comment Management"])
+async def get_comment(comment_id: str, customer_guid: UUID, ticket_id: str):
+    """Retrieve a comment by ID"""
+    try:
+        comment = db_manager.get_comment_by_id(comment_id, str(customer_guid), ticket_id)
+
+        if comment is None:
+            logger.info(f"Comment with comment_id {comment_id} not found for customer {customer_guid}")
+            raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail=f"Comment with comment_id {comment_id} not found for customer {customer_guid}")
+        else:
+            return comment
+    except ValueError as e:
+        logger.error(f"Validation error: {e}")
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail=str(e))
+    except HTTPException as e:
+        raise HTTPException(status_code=HTTPStatus.NOT_FOUND,
+                            detail=f"Comment with comment_id {comment_id} not found for customer {customer_guid}")
+    except Exception as e:
+        if "Database connectivity issue" in str(e):
+            logger.error(f"Database error: {e}")
+            raise HTTPException(
+                status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+                detail="The database is currently unreachable. Please try again later."
+            )
+        logger.error(f"Unexpected error: {e}")
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Internal Server Error")
+
+
+@app.get("/tickets/{ticket_id}/comments", response_model=List[CommentByTicketId], tags=["Comment Management"])
```

Comment:
How are you giving the inputs of page and page_size?

it should be like this
/tickets/{ticket_id}/comments?page=2&page_size=40

Refer to get all chats

Here you have only given default values as 1 and 10, but what about other pages and other page sizes?

Reply:
Sir, the request URL is the same as mentioned above